components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/constant/SQLConstants.java

```diff
@@ -114,6 +172,21 @@
             "ATTR_KEY = VALUES(ATTR_KEY), ATTR_VALUE = VALUES(ATTR_VALUE)";
     public static final String INSERT_OR_UPDATE_ATTRIBUTE_H2 = "MERGE INTO IDN_CONFIG_ATTRIBUTE KEY(ID) VALUES(?, ?, "
             + "?, ?)";
+    public static final String INSERT_OR_UPDATE_ATTRIBUTE_POSTGRESQL = "INSERT INTO IDN_CONFIG_ATTRIBUTE" +
+            "(ID, RESOURCE_ID, ATTR_KEY, ATTR_VALUE) VALUES(?, ?, ?, ?) ON CONFLICT(ID) " +
+            "DO UPDATE SET RESOURCE_ID = EXCLUDED.RESOURCE_ID, ATTR_KEY = EXCLUDED.ATTR_KEY, " +
+            "ATTR_VALUE = EXCLUDED.ATTR_VALUE";
+    public static final String INSERT_OR_UPDATE_ATTRIBUTE_DB2 = "MERGE INTO IDN_CONFIG_ATTRIBUTE USING " +
+            "\"SYSIBM\".DUAL ON (ID= ?) WHEN MATCHED THEN UPDATE SET RESOURCE_ID = ?, ATTR_KEY = ?, ATTR_VALUE = ? " +
+            "WHEN NOT MATCHED THEN INSERT (ID, RESOURCE_ID, ATTR_KEY, ATTR_VALUE) VALUES(?, ?, ?, ?)";
+    public static final String INSERT_OR_UPDATE_ATTRIBUTE_MSSQL = "MERGE IDN_CONFIG_ATTRIBUTE T USING " +
+            "(VALUES  (?, ?, ?, ?)) S (ID, RESOURCE_ID, ATTR_KEY, ATTR_VALUE) ON T.ID = S.ID " +
+            "WHEN MATCHED THEN UPDATE SET RESOURCE_ID = S.RESOURCE_ID, ATTR_KEY = S.ATTR_KEY," +
+            "ATTR_VALUE = S.ATTR_VALUE WHEN NOT MATCHED THEN INSERT (ID, RESOURCE_ID, ATTR_KEY, ATTR_VALUE) " +
+            "VALUES (S.ID, S.RESOURCE_ID, S.ATTR_KEY, S.ATTR_VALUE);";
+    public static final String INSERT_OR_UPDATE_ATTRIBUTE_ORACLE = "MERGE INTO IDN_CONFIG_ATTRIBUTE USING dual " +
+            "ON (ID = ?) WHEN MATCHED THEN UPDATE SET RESOURCE_ID=?, ATTR_KEY=? , ATTR_VALUE=? " +
+            "WHEN NOT MATCHED THEN INSERT (ID,RESOURCE_ID,ATTR_KEY,  ATTR_VALUE) VALUES (?,?, ?, ?)";
```

Comment:
add space after ,

Reply:
addressed in 5a0504e41b864eb30d7aa555f60249f9b40e79e9